ca_gazebo/include/ca_gazebo/color_sensor_plugin.h

```diff
@@ -0,0 +1,64 @@
+    #ifndef GAZEBO_ROS_COLOR_SENSOR_HH
+    #define GAZEBO_ROS_COLOR_SENSOR_HH
+    
+    #include <string>
+    
+    // library for processing camera data for gazebo / ros conversions
+    #include <gazebo/plugins/CameraPlugin.hh>
+
+
+    #include <gazebo_plugins/gazebo_ros_camera_utils.h>
+    
+
+    namespace gazebo
+    {
+      class GazeboRosColor : public CameraPlugin, GazeboRosCameraUtils
```

Comment:
Does this work if you're not subscribed to the camera topic?

Reply:
Hint: it seems so, check [here](https://bitbucket.org/osrf/gazebo/src/default/plugins/CameraPlugin.cc). If not, you should inherit from `SensorPlugin`.

Reply:
It works